src/main/java/com/internet/shop/dao/jdbc/OrderDaoJdbcImpl.java

```diff
@@ -0,0 +1,158 @@
+            return order;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Creation order " + order + " failed", e);
+        }
+    }
+
+    @Override
+    public Optional<Order> getById(Long id) {
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(
+                    "SELECT * FROM orders o "
+                    + "JOIN orders_products op ON o.id = op.order_id "
+                    + "JOIN products p on op.product_id = p.id "
+                    + "WHERE o.id = ? AND o.deleted = FALSE");
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                return Optional.of(getOrderFromResultSet(resultSet));
+            }
+            return Optional.empty();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Getting order with id " + id + " failed", e);
+        }
+    }
+
+    @Override
+    public List<Order> getUserOrders(Long userId) {
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(
+                    "SELECT * FROM orders o "
+                    + "JOIN orders_products op ON o.id = op.order_id "
+                    + "JOIN products p on op.product_id = p.id "
+                    + "WHERE o.user_id = ? AND o.deleted = FALSE",
+                    ResultSet.TYPE_SCROLL_INSENSITIVE, ResultSet.CONCUR_UPDATABLE);
+            statement.setLong(1, userId);
+            return getOrdersListFromStatement(statement);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Getting orders for user with id "
+                                              + userId + " failed", e);
+        }
+    }
+
+    @Override
+    public List<Order> getAll() {
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(
+                    "SELECT * FROM orders o "
+                    + "JOIN orders_products op ON o.id = op.order_id "
+                    + "JOIN products p on op.product_id = p.id "
+                    + "WHERE o.deleted = FALSE ORDER BY o.id",
+                    ResultSet.TYPE_SCROLL_INSENSITIVE, ResultSet.CONCUR_UPDATABLE);
+            return getOrdersListFromStatement(statement);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can not get items from DB", e);
+        }
+    }
+
+    @Override
+    public Order update(Order order) {
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            deleteProductsFromDB(order.getId(), connection);
+            addProductsToDB(order, connection);
+            PreparedStatement statement = connection.prepareStatement(
+                    "UPDATE orders SET user_id = ? "
```

Comment:
do not think we can change user for order. Is it possible?

Reply:
In the real shop it's really difficult to imagine thise situation. But I tried to update all the fields. If mentor doesn't mind I will remove extra logic.